src/game.cpp

```diff
@@ -4050,15 +4052,34 @@ Creature *game::is_hostile_nearby()
     return is_hostile_within( distance );
 }
 
-Creature *game::is_hostile_very_close()
+Creature *game::is_hostile_very_close( bool dangerous )
 {
-    return is_hostile_within( DANGEROUS_PROXIMITY );
+    return is_hostile_within( DANGEROUS_PROXIMITY, dangerous );
 }
 
-Creature *game::is_hostile_within( int distance )
+Creature *game::is_hostile_within( int distance, bool dangerous )
 {
     for( auto &critter : u.get_visible_creatures( distance ) ) {
         if( u.attitude_to( *critter ) == Creature::Attitude::HOSTILE ) {
+            if( dangerous ) {
+                if( critter->has_flag( MF_RANGED_ATTACKER ) ) {
```

Comment:
I think the check if `Creature` is ranged deserves it's own method inside the `Creature`.

Reply:
Done in https://github.com/CleverRaven/Cataclysm-DDA/pull/45752/commits/c7835f97a37f93a0739b20a50b651fee5b733ff0.